tests/unit/test_parameters/test_bpx.py

```diff
@@ -447,40 +417,35 @@ def test_bpx_user_defined(self):
             "User-defined parameter data function": "x**2",
         }
 
-        filename = "tmp.json"
-        with tempfile.NamedTemporaryFile(
-            suffix=filename, delete=False, mode="w"
-        ) as tmp:
-            json.dump(bpx_obj, tmp)
-            tmp.flush()
+        temp_file = tmp_path / "tmp.json"
+        temp_file.write_text(json.dumps(bpx_obj))
 
-            param = pybamm.ParameterValues.create_from_bpx(tmp.name)
+        param = pybamm.ParameterValues.create_from_bpx(str(temp_file))
 
-            assert param["User-defined scalar parameter"] == pytest.approx(
-                1.0, rel=1e-12
-            )
-            var = pybamm.Variable("var")
-            assert isinstance(
-                param["User-defined parameter data"](var), pybamm.Interpolant
-            )
-            assert isinstance(
-                param["User-defined parameter data function"](var), pybamm.Power
-            )
+        assert param["User-defined scalar parameter"] == pytest.approx(1.0, rel=1e-12)
+        var = pybamm.Variable("var")
+        assert isinstance(param["User-defined parameter data"](var), pybamm.Interpolant)
+        assert isinstance(
+            param["User-defined parameter data function"](var), pybamm.Power
+        )
 
-    def test_bpx_activation_energy_default(self):
+    def test_bpx_activation_energy_default(self, tmp_path):
         bpx_obj = copy.deepcopy(self.base)
         del bpx_obj["Parameterisation"]["Negative electrode"][
             "Diffusivity activation energy [J.mol-1]"
         ]
-        with tempfile.NamedTemporaryFile(
-            suffix="test.json", delete=False, mode="w"
-        ) as test_file:
-            json.dump(copy.deepcopy(bpx_obj), test_file)
-            test_file.flush()
-            param = pybamm.ParameterValues.create_from_bpx(test_file.name)
-            assert param[
-                "Negative electrode diffusivity activation energy [J.mol-1]"
-            ] == pytest.approx(0.0, rel=1e-12)
+
+        test_file = tmp_path / "test.json"
+        test_file.write_text(json.dumps(bpx_obj))
+
+        # Ensure the file is actually written
+        assert test_file.exists(), f"File {test_file} was not created."
```

Comment:
I'm curious, do we need this check? I think it is `pytest`'s job to ensure that the file is created, plus `pybamm.ParameterValues.create_from_bpx(str(test_file))` below will fail anyway if it fails to find the file for any reason.